mitiq/executor/executor.py

```diff
@@ -271,14 +271,12 @@ def run(
                 batch = to_run[i * step : (i + 1) * step]
                 self._call_executor(batch, **kwargs)
 
-        these_results = self._quantum_results[start_result_index:]
+        results = self._quantum_results[start_result_index:]
 
-        if force_run_all:
-            return these_results
-
-        # Expand computed results to all results using counts.
-        results_dict = dict(zip(collection.keys(), these_results))
-        results = [results_dict[key] for key in hashable_circuits]
+        if not force_run_all:
+            # Expand computed results to all results using counts.
+            results_dict = dict(zip(collection.keys(), results))
+            results = [results_dict[key] for key in hashable_circuits]
```

Comment:
It looks like this change modifies the return type in what is now the default case of `force_run_all=True` since now everything is going through `self._post_run`. Maybe `_post_run` doesn't do anything in the `force_run_all=True` case, though? Can you clarify slightly?

Reply:
The scope of `_post_run` is to always post-process the measurement results, independently on the value of `force_run_all`. 

What you noticed is correct: before this PR,  `_post_run` was not called in the case `force_run_all=True`. But that was wrong :smile: !   It's a proof that using multiple `return` statements is handy but can be dangerous.